modules/xfem/src/dirackernels/XFEMPressure.C

```diff
@@ -0,0 +1,94 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "XFEMPressure.h"
+#include "Function.h"
+#include "GeometricSearchData.h"
+#include "ElementPairLocator.h"
+#include "FEProblem.h"
+#include "DisplacedProblem.h"
+
+template<>
+InputParameters validParams<XFEMPressure>()
+{
+  InputParameters params = validParams<DiracKernel>();
+  params.addRequiredParam<unsigned int>("component", "The component for the pressure");
+  params.addParam<Real>("factor", 1.0, "The magnitude to use in computing the pressure");
+  params.addParam<FunctionName>("function", "The function that describes the pressure");
+  params.addParam<bool>("use_displaced_mesh", false, "Whether to use geometric search on displaced mesh.");
+  return params;
+}
+
+XFEMPressure::XFEMPressure(const InputParameters & parameters) :
+  DiracKernel(parameters),
+  _component(getParam<unsigned int>("component")),
+  _factor(getParam<Real>("factor")),
+  _function(isParamValid("function") ? &getFunction("function") : NULL),
+  _displaced(getParam<bool>("use_displaced_mesh"))
+{
+  FEProblem * fe_problem = dynamic_cast<FEProblem *>(&_subproblem);
+
+  if (fe_problem == NULL)
+    mooseError("Problem casting _subproblem to FEProblem in XFEMPressure");
+
+  if (!_displaced)
+  {
+    GeometricSearchData & geom_search_data = fe_problem->geomSearchData();
+    element_pair_locators = &geom_search_data._element_pair_locators;
+  }
+  else
+  {
+    GeometricSearchData & displaced_geom_search_data = fe_problem->getDisplacedProblem()->geomSearchData();
+    element_pair_locators = &displaced_geom_search_data._element_pair_locators;
```

Comment:
Per our discussion today, we are not yet setting up any element pair locators on the displaced problem, so this would fail right now if you use the displaced mesh. It will be easy to set that up, so we should just do that and add a test with the displaced option.